### ISSUE TITLE
Music Player stops automatically after playing for some time and using some apps .

### ISSUE THREAD

**Issue by iampawan** [marverenic/Jockey#112]

**Comment by marverenic** [comment]
This sounds like a low memory condition where background apps are being killed to allocate more RAM to the foreground application. I've experienced similar behavior while playing Pokémon GO with Jockey running in the background on my 2nd-generation Moto X (even though it has 2GB of RAM).

When this happens, what behavior do you see? I've only seen this a couple of times and to the best of my knowledge, Jockey wasn't running at all after playback stopped – the service and its notification were completely stopped by the system and weren't restarted. You can also tell that this is the case because recent apps will take a while to restart – the most notable being the launcher since it takes a couple of seconds to redraw itself.

I suspect that this isn't an issue that Jockey can actually work around. Android has very few special cases for determining which apps get pushed out of RAM first (because otherwise developers would take advantage of this too much), and Jockey is already one of the last apps that will get pushed out of RAM because it has an ongoing foreground service. Because the service also runs in its own process, Jockey can continue running in the background using relatively little RAM (10-20MB). All of these factors combined make it very undesirable for Android to kill Jockey, meaning that one of your apps is really pressuring the OS for more memory (which is especially common of games and web browsers).

**Comment by iampawan** [comment]
You are right and it is fine.  But sometimes it closes again and again with
only 2-4 apps working simultaneously and suppose i started song A then B
plays then C and it stops in the background.  Then when we open the app it
again starts with A not where we left i.e.  C?

On 12-Aug-2016 10:11 pm, "Andrew Bailey" <notifications@github.com> wrote:

This sounds like a low memory condition where background apps are being
killed to allocate more RAM to the foreground application. I've experienced
similar behavior while playing Pokémon GO with Jockey running in the
background on my 2nd-generation Moto X (even though it has 2GB of RAM).

When this happens, what behavior do you see? I've only seen this a couple
of times and to the best of my knowledge, Jockey wasn't running at all
after playback stopped – the service and its notification were completely
stopped by the system and weren't restarted. You can also tell that this is
the case because recent apps will take a while to restart – the most
notable being the launcher since it takes a couple of seconds to redraw
itself.

I suspect that this isn't an issue that Jockey can actually work around.
Android has very few special cases for determining which apps get pushed
out of RAM first (because otherwise developers would take advantage of this
too much), and Jockey is already one of the last apps that will get pushed
out of RAM because it has an ongoing foreground service. Because the
service also runs in its own process, Jockey can continue running in the
background using relatively little RAM (10-20MB). All of these factors
combined make it very undesirable for Android to kill Jockey, meaning that
one of your apps is really pressuring the OS for more memory (which is
especially common of games and web browsers).

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/marverenic/Jockey/issues/112#issuecomment-239496757>,
or mute the thread

**Comment by marverenic** [comment]
Jockey saves its player state in the service's `onDestroy()` method and when the service is manually dismissed. When the system kills the app in a low memory condition `onDestroy()` is not called, so this state isn't saved (You'll also notice this behavior if Jockey's player service crashes or gets manually killed).

There are a couple of ways to work around this behavior – the obvious one is just to save the state more often. The only problem with here is that there's a tradeoff in doing this since it requires disk writes and a computation to serialize the state of the player.

**Comment by iampawan** [comment]
Okay..  And are you working on any new update?

**Comment by marverenic** [comment]
Of course :wink:. The best place to check for updates is on Jockey's [Pivotal Tracker](https://www.pivotaltracker.com/n/projects/1594253) page. It's basically the project's entire roadmap including almost all of the open issues on GitHub. Everything's prioritized and estimated for upcoming releases, but all of the deadlines are flexible (yet realistic) since it's a side project.

**Comment by iampawan** [comment]
Yeah i checked but it is flexible that's why i asked.  Anyway great work.
Always there for you.  Let me know if i can do anything for you anytime.
Waiting eagerly for new updates and features.  Thanks bro
